server/text_generation_server/models/causal_lm.py

```diff
@@ -630,6 +630,11 @@ def __init__(
 
         self.enable_hpu_graph = os.getenv("ENABLE_HPU_GRAPH", "true").lower() == "true"
         self.limit_hpu_graph = os.getenv("LIMIT_HPU_GRAPH", "false").lower() == "true"
+
```

Comment:
this should be in class starcoder contructor, if statement won't be necessary then

Reply:
Good idea, will move it, thanks.

Reply:
@kdamaszk , would like your opinion on this alternative: Instead of the assert (which prints this stack trace that looks like it's not handled cleanly), how about we disable hpu graphs and print a warning instead? This will also make it so customers won't have to run the same thing twice i.e run, see it fail and disable hpu graphs and run again.

Reply:
Sounds good!